storage/block.go

```diff
@@ -93,9 +100,12 @@ func (b *dbBlock) Stream(blocker m3db.Context) (m3db.SegmentReader, error) {
 	if blocker != nil {
 		b.ctx.DependsOn(blocker)
 	}
-	s := b.stream
+	var s m3db.SegmentReader
 	if b.writable {
 		s = b.encoder.Stream()
+	} else if b.segment != nil {
```

Comment:
Shouldn't we either be writable _or_ have a `b.segment`?  Perhaps better to just assume and not even bother with the `else if` to make it more clear for readers of the code.